src/finch/interface/lazy.py

```diff
@@ -959,6 +1002,419 @@ def vecdot(x1, x2, /, *, axis=-1) -> LazyTensor:
+            f"Tensor with shape {tensor.shape} is not broadcastable "
+            f"to the shape {shape}"
+        )
+    return elementwise(first_arg, tensor, NoneTensor(shape))
+
+
+def broadcast_arrays(*arrays: LazyTensor) -> tuple[LazyTensor, ...]:
+    """
+    Broadcasts one or more arrays against one another.
+    """
+    shape = _broadcast_shape(*(array.shape for array in arrays))
+    return tuple(broadcast_to(arr, shape) for arr in arrays)
+
+
+class ConcatTensor(Tensor):
+    """
+    Tensor representing a concatenation of multiple tensors along a specified axis.
+    """
+
+    def __init__(self, tensor, *tensors, axis: int = 0):
+        """
+        Args:
+            tensor (ArrayLike):
+                The first tensor.
+            *tensors (ArrayLike):
+                Tensors to concatenate with the first tensor.
+        All tensors must support `__getitem__` and have a `shape` attribute.
+        `fill_value` is taken from the first tensor.
+        `element_type` is casted according to array_api specification.
+        """
+        self._ndim = len(tensor.shape)
+
+        shape_without_axis = tensor.shape[:axis] + tensor.shape[axis + 1 :]
+        # keep track of partial sums of sizes along the concatenation axis
+        self.ps_sizes = [0, tensor.shape[axis]]
+        for t in tensors:
+            if t.ndim != self._ndim:
+                raise ValueError("All tensors must have same number of dimensions")
+            self.ps_sizes.append(self.ps_sizes[-1] + t.shape[axis])
+            if t.shape[:axis] + t.shape[axis + 1 :] != shape_without_axis:
+                raise ValueError(
+                    "All tensors must have the same shape except "
+                    "along the concatenation axis"
+                )
+        self._shape = (
+            tensor.shape[:axis] + (self.ps_sizes[-1],) + tensor.shape[axis + 1 :]
+        )
+        self.tensors = (tensor,) + tensors
+        self.concat_axis = axis
+        # find the appropriate element type
+        self._element_type = element_type(tensor)
+        for t in tensors:
+            self._element_type = promote_type(element_type(self), element_type(t))
+
+    def __getitem__(self, idxs: tuple[int, ...]):
+        """
+        Args:
+            idxs: tuple[int, ...]
+                Indices to access the concatenated tensor.
+        Returns the element at the specified indices.
+        """
+        # find the tensor to access
+        tn = bisect.bisect(self.ps_sizes, idxs[self.concat_axis]) - 1
+        if tn < 0 or tn >= len(self.tensors):
+            raise IndexError(f"Index {idxs} out of bounds for shape {self.shape}")
+        t = self.tensors[tn]
+        shifted_idx = idxs[self.concat_axis] - self.ps_sizes[tn]
+        result = t[
+            idxs[: self.concat_axis] + (shifted_idx,) + idxs[self.concat_axis + 1 :]
+        ]
+        return self.format.element_type(result)
+
+    @property
+    def format(self):
+        return LazyTensorFormat(
```

Comment:
make a separate format called "ConcatTensorFormat". It should contain  the formats of the tensors being concatted